File: C# Advanced/C# OPP/04.Inheritance - Exercise/Inheritance - Exercise/01.Person/Person.cs
using System;

namespace Person
{
    public class Person
    {
        private int age;
        public Person(string name, int age)
        {
            this.Name = name;
            this.Age = age;
            
        }

        public string Name { get; set; }

        public int Age
        {
            get
            {
                return this.age;
            }
            set
            {
                if (value < 0)
                {
                    throw new ArgumentException("People should not be able to have a negative age");
                }
                this.age = value;
            }
        }

        public override string ToString()
        {
            return $"Name: {this.Name}, Age: {this.Age}";
        }
    }
}

File: PHP/PHP Basics/14.Defining Classes -Exercises/Person.php
<?php

class Person
{
    private $name;
    private $age;

    public function __construct($name, $age)
    {
        $this->name = $name;
        $this->age = $age;
    }

    /**
     * @return mixed
     */
    public function getAge()
    {
        return $this->age;
    }

    /**
     * @return mixed
     */
    public function getName()
    {
        return $this->name;
    }

    public function __toString()
    {
        return $this->getName() . ' - ' . $this->getAge();
    }
}

$count_lines = intval(readline());
$persons = [];

for ($i = 0; $i < $count_lines; $i++) {
    $person_data = explode(' ', readline());
    $name = $person_data[0];
    $age = intval($person_data[1]);

    if ($age > 30) {
        $person = new Person($name, $age);
        array_push($persons, $person);
    }
}

usort($persons, function ($p1, $p2) {
    return $p1->getName() <=> $p2->getName();
});

foreach ($persons as $person) {
    $age = $person->getAge();
    echo $person.PHP_EOL;
}